scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -1128,6 +1129,29 @@ void terminateJobHandling(final JobId jobId, final Map<String, String> jobGeneri
         }
     }
 
+    private long getRemoveDelayFromGenericInfo(JobId jobId, Map<String, String> jobGenericInfo, boolean isJobWithErrors,
+            long removeDelay) {
+        if (jobGenericInfo != null) {
+            if (jobGenericInfo.containsKey(GENERIC_INFO_REMOVE_DELAY)) {
+                try {
+                    removeDelay = Tools.parsePeriod(jobGenericInfo.get(GENERIC_INFO_REMOVE_DELAY));
+                } catch (Exception e) {
+                    logger.error("Error when parsing generic information " + GENERIC_INFO_REMOVE_DELAY + " for job " +
+                                 jobId, e);
+                }
+            }
+            if (isJobWithErrors && jobGenericInfo.containsKey(GENERIC_INFO_REMOVE_DELAY_ON_ERROR)) {
```

Comment:
Could the two cases both happen? `else if` might be a bit more clear?

Reply:
yes both case can happen. A user can specific both generic info